### PR TITLE
Facets: resetFacets undefined by default

### PR DESCRIPTION
**Pull request by oshi97**
FilterBox allows you to show a reset all button regardless of searchOnChange
as of #839. Facets also needed to be changed for this change, since it's a
wrapper over FilterBox.

Before, setting resetFacets to true would only render
a reset button if searchOnChange is true. Now, since you can ask for a reset
button even if searchOnChange is false, resetFacets defaulting to true will
render a reset button even if searchOnChange is true. 
By default the reset button should render when searchOnChange is false, and not render when searchOnChange is true. This defaulting is taken care of in FilterBox, so Facets should
default resetFacets to undefined to keep with preexisting defaults.

TEST=manual
Test that in Facets,
with resetFacets unset
- searchOnChange: true hides the reset button
- searchOnChange: unset shows the reset button
- searchOnChange: false shows the reset button

---
- searchOnChange: true, resetFacets: true shows the reset button
- searchOnChange: false, resetFacets: false hides the reset button

### CODE REVIEW

**Comment by creotutar** [comment]
Nit:
> By default the reset button should render _when_ searchOnChange is false

File: src/ui/components/filters/facetscomponent.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ class FacetsConfig {
      * If true, show a "reset all" button to reset all facets
      * @type {boolean}
      */
-    this.resetFacets = config.resetFacets === undefined ? true : config.resetFacets;
```
Comment (creotutar) [inline src/ui/components/filters/facetscomponent.js:43]: README change please

Comment (creotutar) [reply src/ui/components/filters/facetscomponent.js:43]: also sounds like we need a README change for resetFilters in FilterBox?

Comment (oshi97) [reply src/ui/components/filters/facetscomponent.js:43]: resetFilters README was already updated, will update facets README

Comment (oshi97) [reply src/ui/components/filters/facetscomponent.js:43]: sorry coulda swore I pushed this, updated!

**Review by creotutar — APPROVED** [review]
💾 💾 💾 LGTM

**Pull request merged**